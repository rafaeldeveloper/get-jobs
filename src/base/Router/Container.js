import Modules from 'modules';
import React from "react";


const useContainer = () => {

    const routes = {};

    
    const getRouteNameByModule = (str) => str.replace(/(?:^|\.?)([A-Z])/g, function (x, y) { return `-${y.toLowerCase()}` }).replace(/^-/, "");

    const getRoutes = () => {
        
        Object.entries(Modules.Private).map(([key, mod]) => {
            const Module = mod;
            const route = getRouteNameByModule(key);
            routes[`/${route}*`] = () => <Secure><Module /></Secure>;
            return Module
        });


        return routes;
    };

    const authenticated  = () => true;


    const Secure = ({ children }) => {

        if (!authenticated()) {
            // unnautenticad Route
        }

        return children
    }



    return {
        getRoutes
    }
}

export default useContainer