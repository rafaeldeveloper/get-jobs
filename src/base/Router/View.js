import React, { Suspense } from 'react';
import Loading  from 'base/Loading';
import useContainer from './Container';
import { useRoutes, useRedirect } from 'hookrouter';


const AppRouter = () => {


    const { getRoutes } = useContainer();

    const routes = getRoutes();

    const routeResult = useRoutes(routes);
    useRedirect('/', '/vacancy/list');



    return <Suspense fallback={<Loading />}>{routeResult}</Suspense>;
}



export default AppRouter;
