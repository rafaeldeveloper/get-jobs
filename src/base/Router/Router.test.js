import React from 'react';
import Router from 'base/Router';
import renderer from 'react-test-renderer';




test('Router renders correctly', () => {

    const component = renderer.create(
        <Router/>
    );

    let tree = component.toJSON();
    
    expect(tree).toMatchSnapshot();

});

