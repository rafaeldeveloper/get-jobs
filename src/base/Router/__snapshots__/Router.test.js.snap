// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router renders correctly 1`] = `
<div
  className="ui centered divided one column grid"
>
  <div
    className="row"
  >
    <div
      className="sixteen wide column"
    >
      <div
        className="ui stackable menu"
      >
        <div
          className="item"
          onClick={[Function]}
        >
          <img
            alt="logo"
            src="nkey.jpg"
          />
        </div>
        <a
          className="item"
          onClick={[Function]}
        >
          Vacancies
        </a>
        <a
          className="right item"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="plus icon"
            onClick={[Function]}
          />
          Adicionar Anuncio
        </a>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      centered={true}
      className="twelve wide column"
    >
      <div
        className="ui active transition visible dimmer"
        onClick={[Function]}
      >
        <div
          className="content"
        >
          <div
            className="ui medium inverted text loader"
          >
            Loading
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
